src/Controllers/GithubApiController.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LTS\WordpressSecurityAdvisoriesRenovator\Controllers;
+
+use Exception;
+use Github\Client;
+use JsonException;
+use RuntimeException;
+
+/**
+ * Class to fetch data from GitHub API via its Client
+ */
+final readonly class GithubApiController
+{
+    /**
+     * @param string $repo_owner Repository owner + next param
+     * @param string $repo_name  Repository name that needs to be renovated
+     * @param Client $client     GitHub API client
+     */
+    public function __construct(protected Client $client, protected string $repo_owner, protected string $repo_name)
+    {
+    }
+
+    /**
+     * Tries to branch `$branch_name` and head it to `$from_branch`.
+     * Throws a RuntimeException if unable to do so.
+     *
+     * @param string $branch_name
+     * @param string $from_branch
+     *
+     * @throws RuntimeException
+     * @return void
+     */
+    public function createBranch(string $branch_name, string $from_branch): void
+    {
+        $ref_name = sprintf('refs/heads/%1$s', $branch_name);
+        $ref_data = [
+            'ref' => $ref_name,
+            'sha' => $this->getBranchSha($from_branch),
+        ];
+
+        $response = $this->client
+            ->api('gitData')
+            ->references()
+            ->create($this->repo_owner, $this->repo_name, $ref_data);
+
+        if (is_array($response) && $response['ref'] === $ref_name) {
+            return;
+        }
+
+        throw new RuntimeException(sprintf('Unable to create branch, GitHub response was: %1$s', $response));
+    }
+
+    /**
+     * Returns branch SHA
+     *
+     * @param string $branch_name
+     *
+     * @return string|null
+     */
+    public function getBranchSha(string $branch_name): ?string
+    {
+        try {
+            $reference = $this->client
+                ->api('gitData')
+                ->references()
+                ->show($this->repo_owner, $this->repo_name, sprintf('heads/%1$s', $branch_name));
+
+            return is_array($reference)
+                ? $reference['object']['sha'] ?? null
+                : null;
+        } catch (Exception $e) {
```

Comment:
catching the exception and returning null looks dangerous, either log it or do not catch since you probably want the script to die in such a case.

Reply:
fixed